Events.php

```diff
@@ -114,6 +115,20 @@ public static function onRestApiAddRules()
         ], 'wiki');
     }
 
+    public static function onLegalExportServiceCollectUserData($event)
+    {
+        if (!$event instanceof \humhub\modules\legal\events\UserDataCollectionEvent) {
```

Comment:
@yurabakhtin Do we need this check here?

Reply:
@luke- We can remove that if module config will be correct for all other modules as `['class' => 'humhub\modules\legal\services\ExportService', 'event' => 'collectUserData',`.
Or we can replace it with `/* @var \humhub\modules\legal\events\UserDataCollectionEvent $event */` in order to keep a declaration for the method `addUserData` in PhpStormeditor.
Or `public static function onLegalExportServiceCollectUserData(\humhub\modules\legal\events\UserDataCollectionEvent $event)`, but in such case I think we may have an error if the module Legal is not installed.

Reply:
@yurabakhtin On my tests there were not error on unknown symbols when there are not used. Then I would go with following style/naming.

```php
use \humhub\modules\legal\events\UserDataCollectionEvent;

class Events {

    public static function onLegalModuleUserDataExport(UserDataCollectionEvent $event)
    {
         //...
    }

}

```

Reply:
@luke- Yes, you are right, it works wirhout errors, done in the commit https://github.com/humhub/wiki/pull/348/commits/922b0a01bb4452e54d0db6192793fc402d4054fd.